modules/delegates.js

```diff
@@ -555,11 +565,26 @@ Delegates.prototype.isLoaded = function () {
 Delegates.prototype.internal = {
 	forgingEnable: function (req, cb) {
 		library.schema.validate(req.body, schema.enableForging, function (err) {
+
 			if (err) {
 				return setImmediate(cb, err[0].message);
 			}
 
-			var keypair = library.ed.makeKeypair(crypto.createHash('sha256').update(req.body.secret, 'utf8').digest());
+			var keypair;
+			if (library.config.forging.force) {
+				keypair = library.ed.makeKeypair(crypto.createHash('sha256').update(req.body.secret, 'utf8').digest());
+			} else {
+				var encryptedSecret, decryptedSecret;
+				if (Array.isArray(library.config.forging.secret)) {
+					encryptedSecret = library.config.forging.secret[0];
```

Comment:
Why we check against `0` index here? This is not flexible and can be misleading, for example:
- user have few delegates and want to enable forging on one which is not first on the list
- user have one delegate, but want to use different key for each remote server

If we want to force users to not have more than one delegate on list - we should add explicit check (validate the list).